vscode.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/bash
+
+flatpak='flatpak-spawn --host flatpak'
+
+# In order for debugging to work, the the Unity VS Code extension needs to be able to find
+# Unity Editor's PID. Furthermore, it should be relatively unique to avoid conflicts.
+
+# The socket to connect to is found by taking the Unity PID modulo 100 and adding 56000:
+# https://github.com/Unity-Technologies/MonoDevelop.Debugger.Soft.Unity/blob/usedForVSCodeRelease/UnityProcessDiscovery.cs#L93
+
+# The "solution" is to grab a random number and use that as the target PID, then spawn a ton of
+# processes before our Unity fake shell script to ensure the PID is what we want.
+
+# 4 comes because bash will start as PID 2 inside the sandbox, and the cp will be PID
+#. Therfore, we need the minimum target PID to be 4.
+
+# An socat is then set up on the Unity end to connect the socket our PID exposes and the socket
+# that Unity Editor is on.
+
+target_pid=$((4 + RANDOM % 100))
+target_port=$((56000 + target_pid))
+unity_pid=$(pidof Unity)
+unity_port=$((56000 + unity_pid % 1000))
+
+read -r -d '' start_vscode <<'EOF'
+target_pid="$1"
+shift
+
+cp /usr/bin/sleep /tmp/Unity
+
+for (( i=4; i < $target_pid; i++ )); do /usr/bin/true; done
+/tmp/Unity infinity &
+
+if ! [[ -d /usr/lib/sdk/dotnet || -d /usr/lib/sdk/mono5 ]]; then
+  zenity --warning --no-wrap --title='dotnet and Mono SDKs are required' \
+    --text="The dotnet and mono SDK extensions are required to use the Unity debugger integration,
+please install them from Flathub."
```

Comment:
Maybe also launch Software in this case, so we could factor the zenity/dbus invocation into a function.